### ISSUE TITLE
Larger mini batch instead of long input frame

### ISSUE THREAD

**Issue by nakosung** [ibab/tensorflow-wavenet#110]
What is the desirable size for input frame? I think mini batch would be beneficial so we can decrease frame size to get larger mini batch.

Isn't it necessary?

**Comment by ibab** [comment]
What's the benefit of cutting the sample into pieces if you are not planning to do something with the batches, like averaging over them?

At the start of each sample, there's an area where the receptive field of the network isn't saturated.
By cutting the sample into many small pieces, you would increase the ratio that this area occupies in each sample.

**Comment by nakosung** [comment]
Most of deep learning algorithms needs i.i.d. input samples. So I hope that multiple speakers or multiple sessions can be averaged to help learning performance. (Isn't it necessary?)

**Comment by ibab** [comment]
If you want to use techniques like batch normalization, yes.
But the code currently doesn't implement those.
You can open an issue about adding batch normalization if you want.

**Comment by nakosung** [comment]
![image](https://cloud.githubusercontent.com/assets/2463571/18955099/e1fa58a8-8690-11e6-8654-2e56569c25bc.png)

Purple : Minibatch size 1, org frame size
Yellow : Minibatch size 4, 1/4 frame

It seems minibatch doesn't help training perf.
